### PR TITLE
Use which to find binaries

### PR DESCRIPTION
**Pull request by Milo123459**
This PR uses the `which` crate to find binaries.

It is fine to unwrap directly after calling the function as it is more descriptive than the standard error we would print (which is current behaviour).

### CODE REVIEW

**Review by coffee-cup — COMMENTED** [review]
What do the errors look like with unwrap vs no-unwrap?

**Comment by Milo123459** [comment]
Without: 
image

With:
image

Which is clearer IMO.

**Comment by coffee-cup** [comment]
I think we should handle the error a bit better. We shouldn't be showing "panic" or `Result::unwrap` to users of Nixpacks. The error should be something like

"Binary `docker` not found"

**Comment by coffee-cup** [comment]
> We could just .expect it and give more context?

Yeah or add a `context` to the result and don't unwrap.

File: tests/docker_run_tests.rs [hunk 1]
```diff
@@ -249,7 +282,11 @@ fn run_postgres() -> Container {
 }
 
 fn run_mysql() -> Container {
-    let mut docker_cmd = Command::new("docker");
+    let mut docker_cmd = Command::new(
+        which::which("docker")
+            .context("The docker binary could not be found")
+            .unwrap(),
+    );
```
[the remark this answers is not in the document]
Comment (Milo123459) [reply tests/docker_run_tests.rs:289]: image

[the remark this answers is not in the document]
Comment (coffee-cup) [reply tests/docker_run_tests.rs:289]: I think we still shouldn't unwrap. The info about `Resuilt::unwrap` is irrelevant to users of Nixpacks

[the remark this answers is not in the document]
Comment (Milo123459) [reply tests/docker_run_tests.rs:289]: How else should we do it? `panic!`?

[the remark this answers is not in the document]
Comment (coffee-cup) [reply tests/docker_run_tests.rs:289]: `bail!` should work and returns an `Err`

[the remark this answers is not in the document]
Comment (Milo123459) [reply tests/docker_run_tests.rs:289]: image

I've got this. It doesn't use `bail`, but uses `context` and `?`

**Review by coffee-cup — APPROVED** [review]

**Pull request closed without merging**